Server.Tests/ParserTests.cs

```diff
@@ -88,5 +90,51 @@ public void Request_With_Body_Should_Return_Correct_Body()
             Assert.AreEqual("1.1", actual.Version);
             Assert.AreEqual("THIS IS THE BODY", actual.Body);
         }
+
+
+
+        [TestMethod]
+        public void Request_With_Uri_Should_Return_Correct_Uri()
```

Comment:
You're testing the whole process like this, tearing the uri appart and recombining it. You could add some more, smaller and more focused tests.